rust/lance-linalg/src/matrix.rs

```diff
@@ -88,6 +79,9 @@ where
             (&mut rng)
                 .sample_iter(Standard)
                 .take(num_columns * num_rows)
+                .collect::<Vec<f32>>()
+                .iter()
+                .map(|&v| T::Native::from_f32(v).unwrap())
                 .collect::<Vec<_>>(),
```

Comment:
Just curious: is the first collect necessary?

Reply:
It just force the f32 type. Could do something more efficient later.